ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplicationMigrationTool.java

```diff
@@ -0,0 +1,537 @@
+      + "-verifyChecksum: Whether the checksum needs to be validated for each file. Can not be used with "
+      + "-dirLevelCheck \n"
+      + "-filters: Comma separated list of filters, Can not be used along with -dirLevelCheck. \n"
+      + "-conf: Semi-Colon separated list of additional configurations in key1=value1;key2=value2 format. \n"
+      + "-queueSize: Queue size for the thread pool executor. Default:200"
+      + "-numThreads: Number of threads for thread pool executor. Default:10"
+      + "-checksumQueueSize: Queue size for the thread pool executor for checksum computation. Default:200"
+      + "-checksumNumThreads: Number of threads for thread pool executor for checksum computation. Default:5"
+      + "-help: Prints the help message.\n"
+      +"Note: The dumpFilePath for a scheduled query can be fetched using the beeline query: \n"
+      + "select * from sys.replication_metrics where policy_name=‘<policy name>’ order by scheduled_execution_id desc;";
+
+  @Override
+  public int run(String[] args) throws Exception {
+    List<String> argsList = new ArrayList<String>(Arrays.asList(args));
+    boolean isHelp = StringUtils.popOption("-help", argsList);
+    if (isHelp) {
+      System.out.println(help);
+      return 0;
+    }
+
+    // Extract extra configs if provided.
+    final String configs = StringUtils.popOptionWithArgument("-conf", argsList);
+    if (configs != null && !configs.isEmpty()) {
+      String[] configList = configs.split(";");
+      for (String confPair : configList) {
+        String[] confArr = confPair.split("=");
+        if (confArr.length != 2) {
+          throw new IOException("Invalid Configuration " + confPair);
+        } else {
+          conf.set(confArr[0], confArr[1]);
+        }
+      }
+    }
+    final String dumpPath = StringUtils.popOptionWithArgument("-dumpFilePath", argsList);
+    if (dumpPath == null || dumpPath.isEmpty()) {
+      throw new UnsupportedOperationException("-dumpFilePath is not specified");
+    }
+    boolean isValidateChecksum = StringUtils.popOption("-verifyChecksum", argsList);
+    boolean checkAtDirLevel = StringUtils.popOption("-dirLevelCheck", argsList);
+    boolean checkAtFileLevel = StringUtils.popOption("-fileLevelCheck", argsList);
+    boolean checkOpenFiles = StringUtils.popOption("-verifyOpenFiles", argsList);
+    String filter = StringUtils.popOptionWithArgument("-filters", argsList);
+    if (!checkAtFileLevel && isValidateChecksum) {
+      throw new UnsupportedOperationException("-verifyChecksum can not be used without fileLevelCheck");
+    }
+
+    List<Pattern> filtersPattern = new ArrayList<>();
+    if (filter != null && !filter.isEmpty()) {
+      String[] filters = filter.split(",");
+      for (String filterEntry : filters) {
+        Pattern pattern = Pattern.compile(filterEntry);
+        filtersPattern.add(pattern);
+      }
+    }
+
+    String queueSizeString = StringUtils.popOptionWithArgument("-queueSize", argsList);
```

Comment:
move the validations to separate method

Reply:
Moved all validation to separate method and extracted couple of code as individual methods as well